front/src/core-logic/domain/discussion/discussion.test.ts

```diff
@@ -0,0 +1,145 @@
+    new DiscussionBuilder().build(),
+  );
+
+  beforeEach(() => {
+    ({ store, dependencies } = createTestStore());
+  });
+
+  it("on discussion fetched missing", () => {
+    expectDiscussionState(defaultStartingDiscussionState);
+
+    store.dispatch(
+      discussionSlice.actions.fetchDiscussionRequested({
+        jwt,
+        discussionId: "missing-discussion-id",
+      }),
+    );
+
+    expectDiscussionState({ isLoading: true });
+
+    feedGatewayWithDiscussionOrError(undefined);
+
+    expectDiscussionState({
+      isLoading: false,
+      discussion: null,
+      fetchError: null,
+    });
+  });
+
+  it("on discussion fetched successfully", () => {
+    expectDiscussionState(defaultStartingDiscussionState);
+
+    store.dispatch(
+      discussionSlice.actions.fetchDiscussionRequested({
+        jwt,
+        discussionId: discussion.id,
+      }),
+    );
+
+    expectDiscussionState({
+      ...defaultStartingDiscussionState,
+      isLoading: true,
+    });
+
+    feedGatewayWithDiscussionOrError(discussion);
+
+    expectDiscussionState({
+      ...defaultStartingDiscussionState,
+      discussion,
+    });
+  });
+
+  it("on discussion fetched failed", () => {
+    expectDiscussionState(defaultStartingDiscussionState);
+
+    store.dispatch(
+      discussionSlice.actions.fetchDiscussionRequested({
+        jwt,
+        discussionId: discussion.id,
+      }),
+    );
+
+    expectDiscussionState({
+      ...defaultStartingDiscussionState,
+      isLoading: true,
+    });
+
+    feedGatewayWithDiscussionOrError(discussionFetchError);
+
+    expectDiscussionState({
+      ...defaultStartingDiscussionState,
+      fetchError: discussionFetchError.message,
+    });
+  });
+
+  it("previous discussion in state removed on discussion fetch requested", () => {
+    expectDiscussionState(defaultStartingDiscussionState);
+
+    store.dispatch(
+      discussionSlice.actions.fetchDiscussionRequested({
+        jwt,
+        discussionId: discussion.id,
+      }),
+    );
+
+    feedGatewayWithDiscussionOrError(discussion);
+
+    expectDiscussionState({
+      ...defaultStartingDiscussionState,
+      discussion,
+    });
+
+    store.dispatch(
+      discussionSlice.actions.fetchDiscussionRequested({
+        jwt,
+        discussionId: discussion.id,
+      }),
+    );
+
+    expectDiscussionState({
+      ...defaultStartingDiscussionState,
+      isLoading: true,
+      discussion: null,
+    });
+  });
+
+  const feedGatewayWithDiscussionOrError = (
+    discussionOrError?: DiscussionReadDto | Error,
+  ) => {
+    discussionOrError instanceof Error
+      ? dependencies.inclusionConnectedGateway.discussion$.error(
+          discussionOrError,
+        )
+      : dependencies.inclusionConnectedGateway.discussion$.next(
+          discussionOrError,
+        );
+  };
+
+  const expectDiscussionState = (discussionState: Partial<DiscussionState>) => {
+    expectObjectsToMatch(store.getState().discussion, discussionState);
```

Comment:
ca serait mieux de passer par les selectors, pour qu'ils soient testés eux aussi.

Tu peux faire un truc du genre : 

```suggestion
   expect(discussionSelectors.isLoading(store.getState())).toBe(expectedDiscussionState.isLoading);
   expect(discussionSelectors.discussion(store.getState())).toBe(expectedDiscussionState.discussion);
   expect(discussionSelectors.isLoading(store.getState())).toBe(expectedDiscussionState.fetchError)
```

Reply:
Sinon on peut faire 3 fonctions différentes (je te laisse le choix des armes 😉 )

```
expectDiscussionToEqual()

expectIsLoadingToBe()

expectFetchErrorToEqual()
```